test/new.jl

```diff
@@ -2821,16 +2821,21 @@ end
         @test isdir(gmp_jll_dir)
         artifacts_toml = joinpath(gmp_jll_dir, "Artifacts.toml")
         @test isfile(artifacts_toml)
-        meta = artifact_meta("GMP", artifacts_toml)
-        @test meta !== nothing
 
-        gmp_artifact_path = artifact_path(Base.SHA1(meta["git-tree-sha1"]))
-        @test isdir(gmp_artifact_path)
+        # These tests require a matching platform artifact for this old version of GMP_jll,
+        # which is not the case on some of our newer platforms.
+        if !(Sys.isapple() && Sys.ARCH != :aarch64)
```

Comment:
```suggestion
        if !(Sys.isapple() && Base.BinaryPlatforms.arch(Base.BinaryPlatforms.HostPlatform()) == "aarch64")
```
may be more robust because in some cases `Sys.ARCH` could be `:arm64`?  Also, this should be done for armv6l, too, right (in case someone is going to run the tests on that board...)?

Reply:
Yeah, I decided to take a deeper approach to fixing these things.